tests/python/test_moe.py

```diff
@@ -149,20 +166,37 @@ def default_tensor_type(dtype=torch.float32, device="cpu"):
     torch.set_default_device(prev_device)
 
 
-def test_llama4_moe():
+def test_llama4_moe_thunderfx():
     config = Config()
 
     # This is much faster than creating the module with CPU float parameters
     # and then doing `.to("cuda").to(torch.bfloat16)`.
     with default_tensor_type(dtype=torch.bfloat16, device="cuda"):
         model = Llama4MoE(config)
 
+    # Without this, `thunderfx` falls back to `inductor` for `_grouped_mm`
+    # as it doesn't have a grad-rule for the same.
+    for p in model.parameters():
+        p.requires_grad_(False)
+
     batch_size, seq_len = 1, 2048
     inp = torch.randn(
         batch_size, seq_len, config.hidden_size, dtype=torch.bfloat16, device="cuda"
     )
-    out = model(inp)
+    expected = model(inp)
+
+    assert expected.size() == (batch_size, seq_len, config.hidden_size)
+    assert expected.dtype == torch.bfloat16
+    assert expected.is_cuda
+
+    tmodel = thunderfx(model, nv_enable_linear=True)
```

Comment:
:blush: realized I made a mistake in my earlier scatter plumbing.
But with https://github.com/Lightning-AI/lightning-thunder/pull/2435, once that's in, we can add `nv_enable_scatter=True` and I'm seeing a single nvfusion region for the entire thing.

:tada: